pygmt/clib/loading.py

```diff
@@ -96,20 +97,27 @@ def clib_names(os_name: str) -> list[str]:
     return libnames
 
 
-def clib_full_names(env=None):
+def clib_full_names(env: Mapping | dict | None = None) -> Iterator[str]:
```

Comment:
`collections.abc.Mapping` includes `dict`, see https://stackoverflow.com/questions/35690572/what-does-isinstance-with-a-dictionary-and-abc-mapping-from-collections-doing. Maybe we can remove the `dict` part?

```suggestion
def clib_full_names(env: Mapping | None = None) -> Iterator[str]:
```